services/src/main/java/org/keycloak/protocol/oidc/endpoints/TokenEndpoint.java

```diff
@@ -960,17 +960,9 @@ public Response permissionGrant() {
 
         if (permissions != null) {
             event.detail(Details.PERMISSION, String.join("|", permissions));
-            for (String permission : permissions) {
-                String[] parts = permission.split("#");
-                String resource = parts[0];
-
-                if (parts.length == 1) {
-                    authorizationRequest.addPermission(resource);
-                } else {
-                    String[] scopes = parts[1].split(",");
-                    authorizationRequest.addPermission(parts[0], scopes);
-                }
-            }
+            String permissionResourceFormat = formParams.getFirst("permission_resource_format");
+            Boolean permissionResourceMatchingUri = Boolean.parseBoolean(formParams.getFirst("permission_resource_matching_uri"));
```

Comment:
Using a `boolean` type is more appropriate I would say. It would also avoid null checks and make the code simpler.

Reply:
Changed.